src/components/Reports/GDPRPermissions/PermissionExpandableRow.js

```diff
@@ -0,0 +1,175 @@
+  const permissionValue = value.toLowerCase();
+  let iconColor =
+    permissionValue === 'none' ? 'grey' : permissionValue === 'true' ? 'green' : 'red';
+
+  return (
+    <div className="dp-icon-wrapper">
+      <span className={`ms-icon icon-lock dp-lock-${iconColor}`} />
+      <FormattedMessage id={`subscriber_gdpr.value_${permissionValue}`} />
+    </div>
+  );
+};
+
+const PermissionExpandableRow = ({
+  field,
+  email,
+  dependencies: {
+    dopplerApiClient,
+    appConfiguration: { reportsUrl },
+    experimentalFeatures,
+  },
+}) => {
+  const [expanded, setExpanded] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [permissions, setPermissions] = useState([]);
+  const intl = useIntl();
+  const _ = (id, values) => intl.formatMessage({ id }, values);
+  const isPermissionHistoryEnabled =
+    experimentalFeatures && experimentalFeatures.getFeature('PermissionHistory');
+
+  useEffect(() => {
+    const fetchData = async () => {
+      if (isPermissionHistoryEnabled) {
+        const fieldName = field.name;
+        const { success, value } = await dopplerApiClient.getSubscriberPermissionHistory({
+          email,
+          fieldName,
+        });
+        if (success) {
+          setPermissions(value.items);
+        }
+        setLoading(false);
+      }
+    };
+    fetchData();
+  }, [dopplerApiClient]);
+
+  /**
+   * Toggle the expanded state of the row
+   */
+  const toggleExpanded = (e) => {
+    e.preventDefault();
+    setExpanded(!expanded);
+  };
+
+  return (
+    <>
+      <tr>
+        <td>
+          <span>
+            {isPermissionHistoryEnabled && (
+              <a
+                href="#"
+                onClick={toggleExpanded}
+                className={`dp-expand-results ${expanded && 'dp-open-results'}`}
+              >
+                <i className="ms-icon icon-arrow-next"></i>
+              </a>
+            )}
+            {field.name}
+          </span>
+        </td>
+        <td>
+          {field.permissionHTML ? (
+            <S.TextColumn dangerouslySetInnerHTML={{ __html: field.permissionHTML }} />
+          ) : (
+            <FormattedMessage id="subscriber_gdpr.empty_html_text" />
+          )}
+        </td>
+        <td>
+          <PermissionValue value={field.value} />
+        </td>
+      </tr>
+
+      {isPermissionHistoryEnabled && (
+        <tr className={`dp-expanded-table ${expanded && 'show'}`}>
+          {loading ? (
+            <>
+              <td />
+              <td>
+                <S.EmptyBox>
+                  <Loading />
+                </S.EmptyBox>
+              </td>
+              <td />
+            </>
+          ) : (
+            <>
+              <td className="dp-latest-results">
+                <FormattedMessage
+                  id="subscriber_gdpr.latest_results"
+                  tagName="span"
+                  values={{
+                    results_amount: permissions.length,
```

Comment:
Warning, `permissions.length` represents how many results do you get.

But suppose that the history has only 3 changes, `permissions.length` will be `3`, but in my opinion, the label should say _"Last 10 results"_ anyway. Please confirm with the person who defined the functionality.

Reply:
Done! resolving conversation...